### DISCUSSION TITLE
Digital Ocean App Deploy

### DISCUSSION THREAD

**Discussion by mar627** [statamic/cms#6065] [Troubleshooting]
Hello,

I have deployed a site without a starter kit using DO apps, pretty neat. However, after reading and following [this heroku article](https://devcenter.heroku.com/articles/php-support#using-optional-extensions) about how to enable statamic required extensions (bcmath, exif, etc.) I seem to still be hitting this error (from the laravel.log) when running composer update:
```
apps@netflix-7fb4f5d6dc-sdtb9:~$ composer update
> Statamic\Console\Composer\Scripts::preUpdateCmd
Loading composer repositories with package information
Info from https://repo.packagist.org: #StandWithUkraine
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires PHP extension ext-bcmath * but it is missing from your system. Install or enable PHP's bcmath extension.
  Problem 2
    - Root composer.json requires PHP extension ext-exif * but it is missing from your system. Install or enable PHP's exif extension.
  Problem 3
    - Root composer.json requires PHP extension ext-gd * but it is missing from your system. Install or enable PHP's gd extension.

To enable extensions, verify that they are enabled in your .ini files:
    - /app/.heroku/php/etc/php/php.ini
    - /app/.heroku/php/etc/php/conf.d/000-heroku.ini
    - /app/.heroku/php/etc/php/conf.d/010-ext-zend_opcache.ini
    - /app/.heroku/php/etc/php/conf.d/020-ext-mbstring.ini
You can also run `php --ini` in a terminal to see which files are used by PHP in CLI mode.
Alternatively, you can run Composer with `--ignore-platform-req=ext-bcmath --ignore-platform-req=ext-exif --ignore-platform-req=ext-gd` to temporarily ignore these required extensions.
apps@netflix-7fb4f5d6dc-sdtb9:~$ 
```
The site does work, and I overcame a small https mixed content issue — but I'm not getting thumbnails on my assets in the cp, must be that php module ext-gd.. grr

I'm not super savvy, any help is appreciated. Thank you!

**Comment by joshuablum** [comment]
Hey @mar627,

Can you share your `composer.json` file? You added the required PHP extensions in there, right? 

Here's a link to a guide specifically on how to deploy Statamic to DO's app platform: https://www.howtocode.io/posts/how-to-deploy-statamic-3-to-digitalocean-app-platform#php-extensions


You should be good adding this to your `composer.json`
```
"require": {
    //...
    "ext-bcmath": "*",
    "ext-exif": "*",
    "ext-gd": "*",
    //...
},
```

Hope that helps :)

**Reply by mar627** [reply]
Hi @joshuablum, thanks for the response! I did actually read your article! And yep I did add the extensions in..
```
    "require": {
        "php": "^8.0",
        "ext-bcmath": "*",
        "ext-exif": "*",
        "ext-gd": "*",
        "fruitcake/laravel-cors": "^2.0.5",
        "guzzlehttp/guzzle": "^7.2",
        "laravel/framework": "^9.0",
        "laravel/sanctum": "^2.14",
        "laravel/tinker": "^2.7",
        "statamic/cms": "3.3.*",
        "statamic/ssg": "^1.0"
    },
```

I thought maybe the process differed with php8, I know the modules are sometimes included/enabled and other times not, so I went for [the heroku optional extensions docs](https://devcenter.heroku.com/articles/php-support#using-optional-extensions) and the method for enabling required modules remained the same as your instruction.. just that a `composer update` was required afterwards..

So after jumping into the DO app console once I'd updated my `composer.json` file, I punched in `composer update` I received the error message.. argh!
